pkg/virt-controller/watch/vm/vm.go

```diff
@@ -810,6 +812,10 @@ func (c *Controller) handleVolumeUpdateRequest(vm *virtv1.VirtualMachine, vmi *v
 		return nil
 	}
 
+	if hasEphemeralHotplugVolumes(vm, vmi) {
+		return nil
+	}
+
```

Comment:
I want to check that my understanding of this is accurate.

Is this if statement saying that we are going to skip processing the rollout of hotplug volumes from the VM to the VMI if the VMI has hotplug volumes that don't exist on the VM?

but we don't consider that scenario an "ephemeral volume" if `UpdateVolumesStrategyHotplug` is in use?

Reply:
Correct, with `UpdateVolumesStrategyHotplug`, say the user removes a previously hotplugged disk+volume from the VM. The VM+VMI will look no different than if someone used the subresource API without `--persist` arg (what we are calling ephemeral hotplug in code) . So with this strategy it is okay to delete that disk+volume from the VMI

I was curious about how ephemeral hotplug volumes are an issue with other volume update strategies though. Maybe @alicefr can help?

Reply:
@mhenriks ephemeral hotplug volumes don't go into the VM, so they aren't considered an update of the VM spec.

Reply:
Is this check really necessary?

Reply:
> @mhenriks ephemeral hotplug volumes don't go into the VM, so they aren't considered an update of the VM spec.

@alicefr Sure but that doesn't mean there aren't updates to the VM spec as well. This code that you added skips checking the VM spec if there are any hotplug volumes

https://github.com/kubevirt/kubevirt/blob/3615492154216f06fa7211c396c00ef0c71dfc8f/pkg/virt-controller/watch/vm/vm.go#L826-L839

> Is this check really necessary?

It is not necessary for `Hotplug` volume update strategy, but I hoping that you can tell me if it is needed for the others

Reply:
@mhenriks it was for fixing the imperative hotplug which changed the volume set but didn't have any strategy set. Now, no I don't think this is necessary since you set the updateVolumeStrategy to `Hotplug`. Although, it is complicated to see if there is an interactions between imperative and declarative hotplug

Reply:
@alicefr should I keep the check in for `Migration` and `Replacement` strategies?

If `Hotplug` volume update strategy is used, we can't support ephemeral volumes in VMI. They will get removed right away because they are not in the VM